### DISCUSSION TITLE
earthkit xarray engine sel() issue

### DISCUSSION THREAD

**Discussion by jeffbask7** [ecmwf/earthkit-data#554] [Q&A]
Selecting coordinates using ds.sel(latitude=y, longitude=x, method='nearest') returns empty data arrays using the earthkit engine. to_xarray(engine = 'cfgrib') does not have this issue. Is this a potential bug?

**Comment by sandorkertesz** [comment]
Thank you for reporting this issue. For my test case it works. How did you generate `ds`? Can you share the input data with me?

**Reply by jeffbask7** [reply]
import earthkit as ekt
ds = ekt.data.from_source("ecmwf-open-data",
        source='aws',
        model='ifs',
        param=["2t"],
        step=[24],
       )
ds_xr_earthkit = ds.to_xarray()
ds_xr_cfgrib = ds.to_xarray(engine='cfgrib')

nearest_value_cfgrib = (ds_xr_cfgrib['t2m'].sel(latitude=35, longitude=-90, method='nearest').item())
nearest_value_earthkit = (ds_xr_earthkit['2t'].sel(latitude=35, longitude=-90, method='nearest').item())

nearest_value_cfgrib is returned as a single value as expected. nearest_value_earthkit generates error due to (ds_xr_earthkit['2t'].sel(latitude=35, longitude=-90, method='nearest').item()) having no value.

**Reply by sandorkertesz** [reply]
Thank you for the reply. I can confirm it is a bug and it happens when there is only a single field in the input. If I use e.g. `step=[24,36]` then

```ds_xr_earthkit['2t'].sel(latitude=35, longitude=-90, method='nearest').values```

works.
The fix will be available in a hotfix release.
